src/modules/launcher/Plugins/Microsoft.Plugin.Uri/UriHelper/ExtendedUriParser.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using Microsoft.Plugin.Uri.Interfaces;
+
+namespace Microsoft.Plugin.Uri.UriHelper
+{
+    public class ExtendedUriParser : IUriParser
+    {
+        public bool TryParse(string input, out System.Uri result)
+        {
+            if (string.IsNullOrEmpty(input))
+            {
+                result = default;
+                return false;
+            }
+
+            //Handle common cases UriBuilder does not handle
+            if (input.EndsWith(":", StringComparison.Ordinal)
+                || input.EndsWith(".", StringComparison.Ordinal)
+                || input.EndsWith(":/", StringComparison.Ordinal))
```

Comment:
Missing test case `://`. Is that case handle by the uri builder?

Reply:
Yes. 

There is a test added for it also

https://github.com/royvou/PowerToys/blob/feature/urlhandler/src/modules/launcher/Plugins/Microsoft.Plugin.Uri.UnitTests/UriHelper/ExtendedUriParserTests.cs#L25